packages/repeater/src/lib/Repeater.ts

```diff
@@ -81,4 +121,59 @@ export class Repeater {
       })
     );
   }
+
+  private setupShutdown(): void {
+    ['SIGTERM', 'SIGINT', 'SIGHUP'].forEach(event =>
+      process.on(event, async () => {
+        if (this.runningStatus === RunningStatus.RUNNING) {
+          try {
+            await this.stop();
+          } catch (e) {
+            this.logger?.error(e.message);
+          }
+        }
+      })
+    );
+  }
+
+  private handleRegisterResult(
+    res: { payload: RegisterRepeaterResult } | undefined
+  ): void {
+    if (!res) {
+      throw new Error('Error registering repeater.');
+    }
```

Comment:
Why don't check this in the `register` method? It should let you make the signature clear and readable.

```suggestion
  private handleRegisterResult(
    res: { payload: RegisterRepeaterResult }
  ): void {
```